### ISSUE TITLE
Potential bug for spot controller: ValueError: Invalid zone ''

### ISSUE THREAD

**Issue by infwinston** [skypilot-org/skypilot#1283]
Not sure why but sometimes `sky spot launch` returned a weird `ValueError: Invalid zone ''` error. and I had to `sky down --purge` the controller to fix it. Needs more investigation.

```
(sky) weichiang@blaze:~/repos/skypilot/examples/tpu-example$ sky spot launch ls
Task from command: ls
Launching a new spot task 'sky-d7cc-weichiang'. Proceed? [Y/n]:
Launching managed spot job sky-d7cc-weichiang from spot controller...
Launching spot controller...
I 10-21 00:47:02 cloud_vm_ray_backend.py:903] To view detailed progress: tail -n100 -f /home/eecs/weichiang/sky_logs/sky-2022-10-21-00-47-02-061621/provision.log
I 10-21 00:47:02 cloud_vm_ray_backend.py:744] Cluster 'sky-spot-controller-ad9397e5' (status: INIT) was previously launched in AWS (us-east-1). Relaunching in that region.
I 10-21 00:47:03 cloud_vm_ray_backend.py:1163] Launching on AWS us-east-1 (us-east-1a,us-east-1b,us-east-1c,us-east-1d,us-east-1e,us-east-1f)
I 10-21 00:47:16 log_utils.py:45] Head node is up.
I 10-21 00:47:47 cloud_vm_ray_backend.py:998] Successfully provisioned or found existing VM.

ValueError: Invalid zone ''
```

**Comment by concretevitamin** [comment]
Is there any stacktrace?

May also check the generated controller ray yaml. Wondering if it's related to a recent back compat change.  Note that the log above shows the controller was in INIT status.

**Comment by infwinston** [comment]
Weirdly there's no stacktrace. I tried again and checked the generated controller ray yaml. it seems normal.

```
(sky) weichiang@blaze:~/repos/skypilot$ sky spot launch ls
Task from command: ls
Launching a new spot task 'sky-30e4-weichiang'. Proceed? [Y/n]: Y
Launching managed spot job sky-30e4-weichiang from spot controller...
Launching spot controller...
I 10-21 11:14:31 cloud_vm_ray_backend.py:903] To view detailed progress: tail -n100 -f /home/eecs/weichiang/sky_logs/sky-2022-10-21-11-14-30-748100/provision.log
I 10-21 11:14:31 cloud_vm_ray_backend.py:744] Cluster 'sky-spot-controller-ad9397e5' (status: INIT) was previously launched in AWS (us-east-1). Relaunching in that region.
I 10-21 11:14:32 cloud_vm_ray_backend.py:1163] Launching on AWS us-east-1 (us-east-1a,us-east-1b,us-east-1c,us-east-1d,us-east-1e,us-east-1f)
I 10-21 11:14:45 log_utils.py:45] Head node is up.
I 10-21 11:15:15 cloud_vm_ray_backend.py:998] Successfully provisioned or found existing VM.

ValueError: Invalid zone ''


(sky) weichiang@blaze:~/repos/skypilot/examples/tpu-example$ cat ~/.sky/spot_controller/sky-30e4-weichiang.yml                                                                       [7/1973]
# The template for the spot controller

name: sky-30e4-weichiang

resources:
  disk_size: 50
# It is now using default CPU instance type hard-coded in code for spot controller,
# i.e. m6i.2xlarge (8vCPUs, 32 GB) for AWS, Standard_D8_v4 (8vCPUs, 32 GB) for Azure, and n1-highmem-8 (8 vCPUs, 52 GB) for GCP.
# This allows users without the credits for some of the clouds available to use managed spot instances.
# TODO(zhwu): Fix this to be able to failvoer across clouds with cheaper instance.
#  instance_type: t3.xlarge

file_mounts:
  ~/.sky/spot_tasks/sky-30e4-weichiang.yaml: /tmp/spot-task-sky-30e4-weichiang-aivw76mf

setup: |
  # Install cli dependencies
  # Not using SkyPilot wheels because the wheel can be cleaned up by another process.
  # TODO(zhwu): Keep the dependencies align with the ones in setup.py
  (pip list | grep boto3 > /dev/null 2>&1 && \
   pip list | grep google-api-python-client > /dev/null 2>&1 ) || \
   pip install boto3 awscli pycryptodome==3.12.0 google-api-python-client google-cloud-storage 2>&1 > /dev/null

  # We do not install azure dependencies for now since our subscription does not support spot instances.
  # pip list | grep azure-cli > /dev/null 2>&1 || \
  #  pip3 install azure-cli==2.31.0 azure-core

  pushd /tmp &>/dev/null &&     gcloud --help > /dev/null 2>&1 ||     { mkdir -p ~/.sky/logs &&     wget --quiet https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-
382.0.0-linux-x86_64.tar.gz > ~/.sky/logs/gcloud_installation.log &&     tar xzf google-cloud-sdk-382.0.0-linux-x86_64.tar.gz >> ~/.sky/logs/gcloud_installation.log &&     rm -rf ~/google-c
loud-sdk >> ~/.sky/logs/gcloud_installation.log  &&     mv google-cloud-sdk ~/ &&     ~/google-cloud-sdk/install.sh -q >> ~/.sky/logs/gcloud_installation.log 2>&1 &&     echo "source ~/goog
le-cloud-sdk/path.bash.inc > /dev/null 2>&1" >> ~/.bashrc &&     source ~/google-cloud-sdk/path.bash.inc >> ~/.sky/logs/gcloud_installation.log 2>&1; } &&     { cp ~/.sky/.sky_gcp_config_de
fault ~/.config/gcloud/configurations/config_default > /dev/null 2>&1 || true; } &&     popd &>/dev/null

  # Internal: disable logging for manually logging into the spot controller for debugging.

run: |
  python -u -m sky.spot.controller \
    ~/.sky/spot_tasks/sky-30e4-weichiang.yaml \
    --job-id $SKY_JOB_ID

envs:
  SKYPILOT_USAGE_USER_ID: ad9397e5


  SKYPILOT_DISABLE_USAGE_COLLECTION: 1

(sky) weichiang@blaze:~/repos/skypilot/examples/tpu-example$ cat /tmp/spot-task-sky-30e4-weichiang-aivw76mf
name: sky-30e4-weichiang

resources:
  use_spot: true
  spot_recovery: FAILOVER
  disk_size: 256

num_nodes: 1

run: ls

envs: {}

file_mounts: {}
```

I should have mentioned that there was another error before this (stacktrace below).
I thought it's a network error and re-run but `Invalid zone` error then popped up. I think they might be related. 

This was run on a RISE server. I met this issue twice so it may not be just transient.

```
(sky) weichiang@blaze:~/repos/skypilot$ sky spot launch examples/tpu-example/sky.yaml
Task from YAML spec: examples/tpu-example/sky.yaml
Launching a new spot task 'sky-7ae6-weichiang'. Proceed? [Y/n]:
I 10-21 00:36:52 execution.py:497] Translating file_mounts with local source paths to SkyPilot Storage...
I 10-21 00:36:52 execution.py:523] Workdir '/home/eecs/weichiang/repos/skypilot/examples/tpu-example' will be synced to cloud storage 'skypilot-workdir-weichiang-d66ffa6b'.
I 10-21 00:36:52 execution.py:594] Uploading sources to cloud storage. See: sky storage ls
I 10-21 00:36:55 storage.py:1225] Created GCS bucket sky-tpu-cifar-5 in US-CENTRAL1 with storage class STANDARD
I 10-21 00:36:59 storage.py:1225] Created GCS bucket skypilot-workdir-weichiang-d66ffa6b in US-CENTRAL1 with storage class STANDARD
Launching managed spot job sky-7ae6-weichiang from spot controller...
Launching spot controller...
I 10-21 00:37:04 cloud_vm_ray_backend.py:903] To view detailed progress: tail -n100 -f /home/eecs/weichiang/sky_logs/sky-2022-10-21-00-37-03-375422/provision.log

Traceback (most recent call last):
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/util/connection.py", line 73, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/httpsession.py", line 448, in send
    urllib_response = conn.urlopen(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connectionpool.py", line 755, in urlopen
    retries = retries.increment(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/util/retry.py", line 507, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/packages/six.py", line 770, in reraise
    raise value
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connectionpool.py", line 699, in urlopen
    httplib_response = self._make_request(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connectionpool.py", line 382, in _make_request
    self._validate_conn(conn)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connectionpool.py", line 1010, in _validate_conn
    conn.connect()
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connection.py", line 358, in connect
    conn = self._new_conn()
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <botocore.awsrequest.AWSHTTPSConnection object at 0x7fa7b56c62b0>: Failed to establish a new connection: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/data/weichiang/miniconda3/envs/sky/bin/sky", line 33, in <module>
    sys.exit(load_entry_point('skypilot', 'console_scripts', 'sky')())
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 179, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/cli.py", line 933, in invoke
    return super().invoke(ctx)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 179, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/cli.py", line 933, in invoke
    return super().invoke(ctx)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 200, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 200, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/cli.py", line 2632, in spot_launch
    sky.spot_launch(dag,
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 200, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/execution.py", line 461, in spot_launch
    _execute(
  File "/home/eecs/weichiang/repos/skypilot/sky/execution.py", line 213, in _execute
    handle = backend.provision(task,
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 200, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 179, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/backends/backend.py", line 49, in provision
    return self._provision(task, to_provision, dryrun, stream_logs,
  File "/home/eecs/weichiang/repos/skypilot/sky/backends/cloud_vm_ray_backend.py", line 1724, in _provision
    config_dict = provisioner.provision_with_retries(
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 200, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/backends/cloud_vm_ray_backend.py", line 1396, in provision_with_retries
    config_dict = self._retry_region_zones(
  File "/home/eecs/weichiang/repos/skypilot/sky/backends/cloud_vm_ray_backend.py", line 918, in _retry_region_zones
    config_dict = backend_utils.write_cluster_config(
  File "/home/eecs/weichiang/repos/skypilot/sky/utils/common_utils.py", line 200, in _record
    return f(*args, **kwargs)
  File "/home/eecs/weichiang/repos/skypilot/sky/backends/backend_utils.py", line 804, in write_cluster_config
    _add_auth_to_cluster_config(cloud, yaml_path)
  File "/home/eecs/weichiang/repos/skypilot/sky/backends/backend_utils.py", line 859, in _add_auth_to_cluster_config
    config = auth.setup_aws_authentication(config)
  File "/home/eecs/weichiang/repos/skypilot/sky/authentication.py", line 109, in setup_aws_authentication
    key_pairs = ec2.describe_key_pairs()['KeyPairs']
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/client.py", line 514, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/client.py", line 921, in _make_api_call
    http, parsed_response = self._make_request(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/client.py", line 944, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/endpoint.py", line 119, in make_request
    return self._send_request(request_dict, operation_model)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/endpoint.py", line 202, in _send_request
    while self._needs_retry(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/endpoint.py", line 354, in _needs_retry
    responses = self._event_emitter.emit(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/hooks.py", line 412, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/hooks.py", line 256, in emit
    return self._emit(event_name, kwargs)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/hooks.py", line 239, in _emit
    response = handler(**kwargs)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/retryhandler.py", line 207, in __call__
    if self._checker(**checker_kwargs):
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/retryhandler.py", line 284, in __call__
    should_retry = self._should_retry(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/retryhandler.py", line 320, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/retryhandler.py", line 363, in __call__
    checker_response = checker(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/retryhandler.py", line 247, in __call__
    return self._check_caught_exception(
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/retryhandler.py", line 416, in _check_caught_exception
    raise caught_exception
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/endpoint.py", line 281, in _do_get_response
    http_response = self._send(request)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/endpoint.py", line 377, in _send
    return self.http_session.send(request)
  File "/data/weichiang/miniconda3/envs/sky/lib/python3.8/site-packages/botocore/httpsession.py", line 477, in send
    raise EndpointConnectionError(endpoint_url=request.url, error=e)
botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "https://ec2.us-east-1.amazonaws.com/"
```

**Comment by concretevitamin** [comment]
I meant the generated yaml under ~/.sky/generated/..., that gets used by Ray. Does that look normal too?

**Comment by infwinston** [comment]
Oh that also looks normal to me. plus I don't think ray up would output `Invalid zone` right. it should be coming from skypilot.
```
(sky) weichiang@blaze:~/repos/skypilot$ cat ~/.sky/generated/sky-spot-controller-ad9397e5.yml
cluster_name: sky-spot-controller-ad9397e5

max_workers: 0

upscaling_speed: 0

idle_timeout_minutes: 60

provider:
  type: external
  module: sky.skylet.providers.aws.AWSNodeProvider
  region: us-east-1
  availability_zone: us-east-1a,us-east-1b,us-east-1c,us-east-1d,us-east-1e,us-east-1f
  cache_stopped_nodes: true
  security_group:
    GroupName: sky-sg-weichiang-c547

auth:
  ssh_user: ubuntu

available_node_types:
  ray.head.default:
    resources: {}
    node_config:
      InstanceType: m6i.2xlarge
      ImageId: ami-0729d913a335efca7
      BlockDeviceMappings:
      - DeviceName: /dev/sda1
        Ebs:
          VolumeSize: 50
          VolumeType: gp3
          Iops: 3000
          Throughput: 125

head_node_type: ray.head.default

file_mounts:
  ~/.sky/.runtime_files: /tmp/tmp0tmk8a29

rsync_exclude: []

initialization_commands: []

setup_commands:                                                                                                                                                                     [24/1853]
- (mkdir -p ~/.sky && cp -r ~/.sky/.runtime_files/sky-spot-controller-ad9397e5.yml
  ~/.sky/sky_ray.yml) && (mkdir -p ~/.sky/wheels/3eb2a1d02efc3898c0441810c08c9b5a
  && cp -r ~/.sky/.runtime_files/3eb2a1d02efc3898c0441810c08c9b5a/* ~/.sky/wheels/3eb2a1d02efc3898c0441810c08c9b5a)
  && (mkdir -p ~/.aws && cp -r ~/.sky/.runtime_files/credentials ~/.aws/credentials)
  && (mkdir -p ~/.azure && cp -r ~/.sky/.runtime_files/azureProfile.json ~/.azure/azureProfile.json)
  && (mkdir -p ~/.azure && cp -r ~/.sky/.runtime_files/clouds.config ~/.azure/clouds.config)
  && (mkdir -p ~/.azure && cp -r ~/.sky/.runtime_files/config ~/.azure/config) &&
  (mkdir -p ~/.azure && cp -r ~/.sky/.runtime_files/msal_token_cache.json ~/.azure/msal_token_cache.json)
  && (mkdir -p ~/.config/gcloud && cp -r ~/.sky/.runtime_files/credentials.db ~/.config/gcloud/credentials.db)
  && (mkdir -p ~/.config/gcloud && cp -r ~/.sky/.runtime_files/application_default_credentials.json
  ~/.config/gcloud/application_default_credentials.json) && (mkdir -p ~/.config/gcloud
  && cp -r ~/.sky/.runtime_files/access_tokens.db ~/.config/gcloud/access_tokens.db)
  && (mkdir -p ~/.config/gcloud/configurations && cp -r ~/.sky/.runtime_files/configurations/*
  ~/.config/gcloud/configurations) && (mkdir -p ~/.config/gcloud/legacy_credentials
  && cp -r ~/.sky/.runtime_files/legacy_credentials/* ~/.config/gcloud/legacy_credentials)
  && (mkdir -p ~/.config/gcloud && cp -r ~/.sky/.runtime_files/active_config ~/.config/gcloud/active_config)
  && (mkdir -p ~/.sky && cp -r ~/.sky/.runtime_files/.sky_gcp_config_default ~/.sky/.sky_gcp_config_default);
  mkdir -p ~/.ssh; touch ~/.ssh/config; pip3 --version > /dev/null 2>&1 || (curl -sSL
  https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3 get-pip.py && echo
  "PATH=$HOME/.local/bin:$PATH" >> ~/.bashrc); (type -a python | grep -q python3)
  || echo 'alias python=python3' >> ~/.bashrc; (type -a pip | grep -q pip3) || echo
  'alias pip=pip3' >> ~/.bashrc; (which conda > /dev/null 2>&1 && conda init > /dev/null
  && conda config --set auto_activate_base false) || (wget -nc https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
  && bash Miniconda3-latest-Linux-x86_64.sh -b && eval "$(~/miniconda3/bin/conda shell.bash
  hook)" && conda init && conda config --set auto_activate_base true); source ~/.bashrc;
  (pip3 list | grep ray | grep 1.13.0 2>&1 > /dev/null || pip3 install -U ray[default]==1.13.0)
  && mkdir -p ~/sky_workdir && mkdir -p ~/.sky/sky_app; (pip3 list | grep skypilot
  && [ "$(cat ~/.sky/wheels/current_sky_wheel_hash)" == "3eb2a1d02efc3898c0441810c08c9b5a"
  ]) || (pip3 uninstall skypilot -y; pip3 install "$(echo ~/.sky/wheels/3eb2a1d02efc3898c0441810c08c9b5a/skypilot-1.0.0.dev0*.whl)[aws]"
  && echo "3eb2a1d02efc3898c0441810c08c9b5a" > ~/.sky/wheels/current_sky_wheel_hash
  || exit 1); sudo systemctl stop unattended-upgrades; sudo kill -9 `sudo lsof /var/lib/dpkg/lock-frontend
  | awk '{print $2}' | tail -n 1` || true; sudo pkill -9 apt-get; sudo pkill -9 dpkg;
  sudo dpkg --configure -a; sudo bash -c 'rm -rf /etc/security/limits.d; echo "* soft
  nofile 1048576" >> /etc/security/limits.conf; echo "* hard nofile 1048576" >> /etc/security/limits.conf';
  sudo grep -e '^DefaultTasksMax' /etc/systemd/system.conf || (sudo bash -c 'echo
  "DefaultTasksMax=infinity" >> /etc/systemd/system.conf'); sudo systemctl set-property
  user-$(id -u $(whoami)).slice TasksMax=infinity; sudo systemctl daemon-reload; mkdir
  -p ~/.ssh; (grep -Pzo -q "Host \*\n  StrictHostKeyChecking no" ~/.ssh/config) ||
  printf "Host *\n  StrictHostKeyChecking no\n" >> ~/.ssh/config; python3 -c "from
  sky.skylet.ray_patches import patch; patch()" || exit 1; [ -f /etc/fuse.conf ] &&
  sudo sed -i 's/#user_allow_other/user_allow_other/g' /etc/fuse.conf || (sudo sh
  -c 'echo "user_allow_other" > /etc/fuse.conf');

head_start_ray_commands:
- ((ps aux | grep "-m sky.skylet.skylet" | grep -q python3) || nohup python3 -m sky.skylet.skylet
  >> ~/.sky/skylet.log 2>&1 &); ray stop; ray start --disable-usage-stats --head --port=6379
  --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml  ||
  exit 1; which prlimit && for id in $(pgrep -f raylet/raylet); do sudo prlimit --nofile=1048576:1048576
  --pid=$id || true; done;

worker_start_ray_commands: []

head_node: {}

worker_nodes: {}

head_setup_commands: []

worker_setup_commands: []

cluster_synced_files: []

file_mounts_sync_continuously: false
```

**Comment by Michaelvll** [comment]
Could you log into the spot controller and get `~/.sky/spot_tasks/sky-30e4-weichiang.yaml` file? I am thinking probably, the file has `zone: ''` in it.

**Comment by concretevitamin** [comment]
Should this be closed now that #1287 is merged? @infwinston

**Comment by infwinston** [comment]
Yes not sure why it's not automatically closed.
